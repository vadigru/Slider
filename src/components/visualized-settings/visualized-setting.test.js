import React from "react";
import renderer from "react-test-renderer";

import VisualizedSettings from "./visualized-settings.jsx";

it(`Should render VisualizedSettings component`, () => {
  const tree = renderer
    .create(
        <VisualizedSettings
          setWidth={() => {}}
          setHeight={() => {}}
          toggleInfinite={() => {}}
          toggleAutoplay={() => {}}
          changeAutoplayDelay={() => {}}
          toggleIndicators={() => {}}
          toggleArrows={() => {}}
          toggleAdaptive={() => {}}
          toggleAnimatedSwipe={() => {}}
          toggleCaption={() => {}}
          changeSlidesCount={() => {}}
          changeAnimationTime={() => {}}
          animationTime={1000}
          autoplayDelay={3000}
          isInfinite={true}
          isAutoplay={false}
          isIndicators={true}
          isArrows={true}
          isAdaptive={false}
          isAnimatedSwipe={false}
          isCaption={false}
          slidesToShow={1}
          sliderWidth={1200}
          sliderHeight={600}
        />,
        {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
