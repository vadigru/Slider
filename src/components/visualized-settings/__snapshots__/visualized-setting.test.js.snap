// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render VisualizedSettings component 1`] = `
<section
  className="slider-settings"
>
  <div>
    <label
      htmlFor="wrange"
    >
      <span
        className="label-text"
      >
        Width
      </span>
      <input
        id="wrange"
        max="1920"
        min="320"
        name="wrange"
        onChange={[Function]}
        type="range"
        value={1200}
      />
      <span
        className="label-text-output"
      >
        1200
      </span>
    </label>
  </div>
  <div>
    <label
      htmlFor="hrange"
    >
      <span
        className="label-text"
      >
        Height
      </span>
      <input
        id="hrange"
        max="1080"
        min="320"
        name="hrange"
        onChange={[Function]}
        type="range"
        value={600}
      />
      <span
        className="label-text-output"
      >
        600
      </span>
    </label>
  </div>
  <div>
    <label
      htmlFor="animation-time"
    >
      <span
        className="label-text"
      >
        Animation/Speed
      </span>
      <input
        id="animation-time"
        max="3000"
        min="100"
        name="animation-time"
        onChange={[Function]}
        type="range"
        value={1000}
      />
      <span
        className="label-text-output"
      >
        1000
      </span>
    </label>
  </div>
  <div>
    <input
      className="visually-hidden"
      defaultChecked={false}
      id="autoplay"
      name="autoplay"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="label-autoplay"
      htmlFor="autoplay"
    >
      <span
        className="label-text"
      >
        Autoplay/Interval
      </span>
      <input
        disabled={true}
        id="autoplay-delay"
        max="10000"
        min="1000"
        name="autoplay-delay"
        onChange={[Function]}
        type="range"
        value={3000}
      />
      <span
        className="label-text-output"
      >
        3000
      </span>
    </label>
  </div>
  <div>
    <input
      className="visually-hidden"
      defaultChecked={false}
      id="adaptive"
      name="adaptive"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="label-adaptive"
      htmlFor="adaptive"
    >
      <span
        className="label-text"
      >
        Adaptive/Slides
      </span>
      <input
        disabled={true}
        id="slides-count"
        max="3"
        min="1"
        name="slides-count"
        onChange={[Function]}
        type="range"
        value={1}
      />
      <span
        className="label-text-output"
      >
        1
      </span>
    </label>
  </div>
  <div>
    <input
      className="visually-hidden"
      defaultChecked={false}
      id="animated-swipe"
      name="animated-swipe"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="animated-swipe"
    >
      <span
        className="label-text"
      >
        Animated swipe
      </span>
    </label>
  </div>
  <div>
    <input
      className="visually-hidden"
      defaultChecked="checked"
      id="indicators"
      name="indicators"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="indicators"
    >
      <span
        className="label-text"
      >
        Indicators
      </span>
    </label>
  </div>
  <div>
    <input
      className="visually-hidden"
      defaultChecked="checked"
      id="arrows"
      name="arrows"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="arrows"
    >
      <span
        className="label-text"
      >
        Arrows
      </span>
    </label>
  </div>
  <div>
    <input
      className="visually-hidden"
      defaultChecked={false}
      id="caption"
      name="caption"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="caption"
    >
      <span
        className="label-text"
      >
        Caption
      </span>
    </label>
  </div>
  <div>
    <input
      className="visually-hidden"
      defaultChecked="checked"
      id="infinite"
      name="infinite"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="infinite"
    >
      <span
        className="label-text"
      >
        Infinte
      </span>
    </label>
  </div>
</section>
`;
